OpenRA.Game/Traits/World/ScreenMap.cs

```diff
@@ -27,99 +27,149 @@ public class ScreenMapInfo : ITraitInfo
 
 	public class ScreenMap : IWorldLoaded
 	{
-		ScreenMapInfo info;
+		class Partitioned<T>
```

Comment:
Can this be pulled out somewhere accessible? I will want to use this from another trait it a future PR.